app/src/test/java/org/simple/clinic/registration/facility/RegistrationFacilitySelectionScreenControllerTest.kt

```diff
@@ -228,29 +229,26 @@ class RegistrationFacilitySelectionScreenControllerTest {
 
   @Test
   fun `when facilities are received then their UI models for facility list should be created`() {
-    val facility1 = TestData.facility(name = "Facility 1")
-    val facility2 = TestData.facility(name = "Facility 2")
-    val facilities = listOf(facility1, facility2)
-    whenever(facilityRepository.facilities()).thenReturn(Observable.just(facilities, facilities))
+    val phcObvious = TestData.facility(name = "PHC Obvious", streetAddress = "Richmond Road", district = "Bangalore Central", state = "Karnataka")
+    val chcNilenso = TestData.facility(name = "CHC Nilenso", district = "Indiranagar", state = "Karnataka")
+    val facilities = listOf(phcObvious, chcNilenso)
+    val searchQuery = ""
+
+    whenever(facilityRepository.facilities(searchQuery)).thenReturn(Observable.just(facilities, facilities))
     whenever(facilityRepository.recordCount()).thenReturn(Observable.just(facilities.size, facilities.size))
     whenever(screenLocationUpdates.streamUserLocation(any(), any(), any())).thenReturn(Observable.just(Unavailable))
 
-    val searchQuery = ""
-    val facilityListItems = emptyList<FacilityListItem>()
-    whenever(listItemBuilder.build(any(), any(), any(), any())).thenReturn(facilityListItems)
+    val expectedFacilityListItems = listOf(
+        FacilityOption(phcObvious, Name.Plain("PHC Obvious"), address = Address.WithStreet(street = "Richmond Road", district = "Bangalore Central", state = "Karnataka")),
+        FacilityOption(chcNilenso, Name.Plain("CHC Nilenso"), address = Address.WithoutStreet(district = "Indiranagar", state = "Karnataka"))
+    )
 
     setupController()
     uiEvents.onNext(ScreenCreated())
+
     uiEvents.onNext(RegistrationFacilityUserLocationUpdated(Unavailable))
-    uiEvents.onNext(RegistrationFacilitySearchQueryChanged(searchQuery))
-
-    verify(listItemBuilder, times(2)).build(
-        facilities = facilities,
-        searchQuery = searchQuery,
-        userLocation = null,
-        proximityThreshold = registrationConfig.proximityThresholdForNearbyFacilities)
-    verify(screen).updateFacilities(facilityListItems, FIRST_UPDATE)
-    verify(screen).updateFacilities(facilityListItems, SUBSEQUENT_UPDATE)
```

Comment:
We were using it earlier to provide a different UI animation if the facilities were being loaded for the first time vs. later. This was adding additional complexity, plus in addition we already have a default set of animations that are provided by the recyclerview.

I felt we could remove it since the complexity in managing this is a lot compared to the difference between the default animations and this one.